### ISSUE TITLE
Serious Problem with controller Axis call. 

### ISSUE THREAD

**Issue by RustyRaptor** [FRC-Utilities/LibDS-C#6]
This problem isn't present in the LabView driverstation. When I use a joystick it seems to not return any value until the joystick is pushed all the way to 1 or -1. I am guessing it's converting the float to an integer so you end up with 0 until you hit 1. I've tried multiple joysticks from different vendors. 

Could this be the function causing it? I am not entirely certain how it works. 

```
float DS_GetJoystickAxis (int joystick, int axis)
{
    if (CFG_GetRobotEnabled() && joystick_exists (joystick)) {
        DS_Joystick* stick = get_joystick (joystick);

        if (stick->num_axes > axis)
            return stick->axes [axis];
    }

    return 0;
}
```

**Comment by alex-spataru** [comment]
Thanks for the report. Do the joystick axes move correctly in the "joysticks" tab of the DS? This is to know if the problem is with SDL/QJoysticks or the LibDS protocol implementation.

**Comment by RustyRaptor** [comment]
They do seem to move correctly in the joysticks tab. speaking of which I tried using the virtual joystick and setting it's range to 99 and it didn't work.

**Comment by jacksondm33** [comment]
Same problem here. I'll take a look to see if I can find the source of the problem.

**Comment by alex-spataru** [comment]
@jacksondm33 Thanks! Let me know if you have any questions about the source code of LibDS.

**Comment by jacksondm33** [comment]
```
uint8_t DS_FloatToByte (const float value, const float max)
{
    if (value != 0 && max != 0 && value <= max) {
        int percent = (int) (value / max) * (0xFF / 2);
        return (uint8_t) (percent & 0xFF);
    }

    return 0;
}
```
The problem is here. When calculating the percent, it converts `value / max` to an integer before multiplying it by `0xFF / 2`.
Fixed in #9

**Comment by RustyRaptor** [comment]
Awesome. Thanks. Now we can stop having to put up with the monster known as labview DS :) . I wish they would legalize this station in competitions honestly.
